src/qinfer/derived_models.py

```diff
@@ -380,6 +381,128 @@ def fisher_information(self, modelparams, expparams):
         )
         return two_outcome_fi * expparams['n_meas']
 
+class GaussianHyperparameterizedModel(DerivedModel):
+    """
+    Model representing a two-outcome model viewed through samples
+    from one of two distinct Gaussian distributions. This model adds four new
+    model parameters to its underlying model, respectively representing the
+    mean outcome conditioned on an underlying 0, the mean outcome conditioned
+    on an underlying 1, and the variance of outcomes conditioned in each case.
+
+    :param qinfer.abstract_model.Model underlying_model: An instance of a two-
+        outcome model to be viewed through Gaussian distributions.
+    """
+
+    def __init__(self, underlying_model):
+        super(GaussianHyperparameterizedModel, self).__init__(underlying_model)
+
+        if not (underlying_model.is_n_outcomes_constant and underlying_model.n_outcomes(None) == 2):
+            raise ValueError("Decorated model must be a two-outcome model.")
+
+    ## PROPERTIES ##
+
+    @property
+    def decorated_model(self):
+        # Provided for backcompat only.
+        return self.underlying_model
+
+    @property
+    def modelparam_names(self):
+        return self.underlying_model.modelparam_names + [
+            r'\mu_0', r'\mu_1',
+            r'\sigma_0^2', r'\sigma_1^2'
+        ]
+    
+    @property
+    def n_modelparams(self):
+        return len(self.modelparam_names)
+    
+    ## METHODS ##
+    
+    def domain(self, expparams):
+        return [RealDomain()] * len(expparams)
+    
+    def are_expparam_dtypes_consistent(self, expparams):
+        return True
+
+    def are_models_valid(self, modelparams):
+        orig_mps = modelparams[:, :-4]
```

Comment:
FYI negative indexes (rather than absolute indexes based on ``underlying_model.n_modelparams``) have occasionally caused me trouble when another decoration model is above. However, I don't think anything in the code base actually cares, and I think other models might use negative indexes too, so don't change it if you don't want to. It mostly comes up when I am manually jiggering things in a notebook, and have many models and many updaters on the go.